macaroonbakery/bakery.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2016 Canonical Ltd.
+# Licensed under the AGPLv3, see LICENCE file for details.
+
+import base64
+import json
+import requests
+import webbrowser
+
+import macaroons
+
+
+def deserialize(json_macaroon):
+    '''Deserialize a JSON macaroon into a macaroon object.
+
+    @param the JSON macaroon to deserialize.
+    @return the deserialized macaroon.
+    '''
+    return macaroons.deserialize(to_base64(json_macaroon))
+
+
+def to_base64(json_macaroon):
+    '''Base 64 encode a json macaroon
+
+    @param the JSON macaroon to encode.
+    @return the encoded macaroon.
+    '''
+    return base64.urlsafe_b64encode(json.dumps(json_macaroon))
+
+
+def add_padding(s):
```

Comment:
Is this function really intended to be used by library clients?
Should this be called _add_padding instead?